### ISSUE TITLE
Feature: hook for custom serialized export-extra ID data

### ISSUE THREAD

**Issue by IARI** [KhronosGroup/glTF-Blender-Exporter#274]
This is a follow up to #248.

The application I am working on uses the gltf Format to host a lot of complex custom data.
In addition to exporting properties that are attached to objects, I have found the need to also export complete custom node trees defined in the blender addon which I am working on.

The most flexible and minimalistic approach to allow this that I could come up with so far is the option for a hook in the JSONEncoder in the gltf exporter, which basically looks like this:

```python
            if hasattr(obj, "_gltf_export_data"):
                    custom_data = obj._gltf_export_data()
                    result["data"] = custom_data

```

My current implementation:
[gltf2_create.py#L41](https://github.com/IARI/glTF-Blender-Exporter/blob/export-hook/scripts/addons/io_scene_gltf2/gltf2_create.py#L41)
Diff:
https://github.com/IARI/glTF-Blender-Exporter/commit/f85caf4134b4bde68e0a949a43c88b6b90defff7

The following is an example based on my application. It allows me to export a complete node tree structure (nodes and links) in my own addon, by defining the `_gltf_export_data` method in custom NodeTree classes:

```python
class MyExampleNodeTree(bpy.types.NodeTree):
    ...

    def _gltf_export_data(self):
        def resolve_link_vals(link, key):
            val = getattr(link, key)
            if hasattr(val, 'identifier'):
                yield key, val.name
                yield key + '_identifier', val.identifier
            else:
                yield key, val.name

        result = {
            "nodes": {i: n._gltf_export_data() for i, n in self.nodes.items()},
            "links": [{k: v for key in EXPORT_LINK_PROPS for k, v in resolve_link_vals(link, key)} for link in
                      self.links if link.is_valid]
        }
        return result
```

I have implemented this hook and also started working on an advanced  documentation for the 'use extras' section of the manual including a link to a separate document, in which I plan to describe the usage in more detail. Xou can find my current drafts here:
[/docs/user.md#export-extras](https://github.com/IARI/glTF-Blender-Exporter/blob/export-hook/docs/user.md#export-extras)
[/docs/export_extras_developer.md](https://github.com/IARI/glTF-Blender-Exporter/blob/export-hook/docs/export_extras_developer.md)

@donmccurdy @emackey what do you think about this?
If you find that this may be interesting for the public, I can make a PR with this feature and the documentation.

**Comment by IARI** [comment]
@donmccurdy  @emackey  any thoughts on this?

**Comment by emackey** [comment]
Hi @IARI, active development has moved to [glTF-Blender-IO](https://github.com/KhronosGroup/glTF-Blender-IO).  There's an issue https://github.com/KhronosGroup/glTF-Blender-IO/issues/7 that talks about exporting not just extras, but entire glTF extensions from a 3rd-party Blender plugin.  That's a capability I'm interested in, but I think it may have to wait for development efforts to progress further.

**Comment by IARI** [comment]
Hi @emackey and thanks.
* Is this repo still maintained?
* Is the other one a complete rewrite - will the features (say from the recently merged Pr's) be in that new exporter as well?
* IS there any source of information, where I can follows these kinds of development better?

**Comment by donmccurdy** [comment]
The new importer/exporter is now included with the Blender 2.8 beta, and we won't be adding further features to this older repository. Similar issue on the new repo here: https://github.com/KhronosGroup/glTF-Blender-IO/issues/7.
